RelationalAI/Services/Rest.cs

```diff
@@ -293,14 +314,14 @@ private async Task<AccessToken> RequestAccessTokenAsync(string host, ClientCrede
             var resp = await RequestHelperAsync("POST", creds.ClientCredentialsUrl, data);
             if (!(resp is string stringResponse))
             {
-                throw new SystemException("Unexpected response type");
+                throw new InvalidResponseException("Unexpected response type, expected a string", resp);
```

Comment:
maybe also include the current response type in the exception message

Reply:
@NRHelmi addressed in the latest commit. Thank you.

Reply:
thank you :pray: